components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx

```diff
@@ -1,35 +1,53 @@
+import React, { useCallback, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { useDrawerContext } from './DrawerContext';
-import { useNavGroupContext } from './NavGroupContext';
-import { usePrevious } from '../hooks/usePrevious';
-import { Theme, useTheme } from '@mui/material/styles';
-import createStyles from '@mui/styles/createStyles';
-import makeStyles from '@mui/styles/makeStyles';
+import { useDrawerContext } from '../DrawerContext';
+import { useNavGroupContext } from '../NavGroupContext';
+import { usePrevious } from '../../hooks/usePrevious';
+import { Theme, useTheme, styled, SxProps } from '@mui/material/styles';
 import List from '@mui/material/List';
 import Collapse from '@mui/material/Collapse';
-import { InfoListItem, InfoListItemProps as BLUIInfoListItemProps } from '../InfoListItem';
+import { InfoListItem, InfoListItemProps as BLUIInfoListItemProps } from '../../InfoListItem';
 import ArrowDropDown from '@mui/icons-material/ArrowDropDown';
 import ChevronRight from '@mui/icons-material/ChevronRight';
 import ExpandMore from '@mui/icons-material/ExpandMore';
-import { NavItemSharedStyleProps, NavItemSharedStylePropTypes, SharedStyleProps, SharedStylePropTypes } from './types';
+import { NavItemSharedStyleProps, NavItemSharedStylePropTypes, SharedStyleProps, SharedStylePropTypes } from '../types';
 import clsx from 'clsx';
 import color from 'color';
-import { findChildByType, mergeStyleProp } from './utilities';
+import { findChildByType, mergeStyleProp } from '../utilities';
 import { white, darkBlack } from '@brightlayer-ui/colors';
-import { DrawerRailItemProps } from './DrawerRailItem';
+import { DrawerRailItemProps } from '../DrawerRailItem';
+import { cx } from '@emotion/css';
+import drawerNavItemClasses, {
+    DrawerNavItemClasses,
+    DrawerNavItemClassKey,
+    getDrawerNavItemUtilityClass,
+} from './DrawerNavItemClasses';
+import Box, { BoxProps } from '@mui/material/Box';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
 
-export type DrawerNavItemClasses = {
-    root?: string;
-    active?: string;
-    expandIcon?: string;
-    infoListItemRoot?: string;
-    nestedListGroup?: string;
-    nestedTitle?: string;
-    title?: string;
-    titleActive?: string;
-    ripple?: string;
+const useUtilityClasses = (ownerState: DrawerNavItemProps): Record<DrawerNavItemClassKey, string> => {
+    const { classes } = ownerState;
+
+    const slots = {
+        root: ['root'],
+        active: ['active'],
+        expandIcon: ['expandIcon'],
+        infoListItemRoot: ['infoListItemRoot'],
+        nestedListGroup: ['nestedListGroup'],
+        nestedTitle: ['nestedTitle'],
+        title: ['title'],
+        titleActive: ['titleActive'],
+        ripple: ['ripple'],
+        expanded: ['expanded'],
+        drawerOpen: ['drawerOpen'],
+        noIconTitle: ['noIconTitle'],
+        textSecondary: ['textSecondary'],
+        square: ['square'],
```

Comment:
Why are these classes being added. The associated styles should be conditionally applied within the `styled` functions. There are plenty examples of this now that you can follow.